### PR TITLE
Removed exoplanets diagram from the GUI

### PR DESCRIPTION
**Pull request by alex-w**
### Description
Removed exoplanets diagram from the GUI of the plugin as an intermediate step to the Qt6 migration process.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by gzotti** [comment]
Why not replace by QtChart immediately? This could probably re-use most of the plotting code.

File: plugins/Exoplanets/src/gui/ExoplanetsDialog.cpp [hunk 1]
```diff
@@ -275,7 +270,7 @@ void ExoplanetsDialog::setAboutHtml(void)
 	html += "<tr><td><strong>" + q_("Author") + ":</strong></td><td>Alexander Wolf</td></tr></table>";
 
 	html += "<p>" + QString(q_("This plugin plots the position of stars with exoplanets. Exoplanets data is derived from \"%1The Extrasolar Planets Encyclopaedia%2\"")).arg("<a href=\"http://exoplanet.eu/\">", "</a>. ");
```
Comment (gzotti) [inline plugins/Exoplanets/src/gui/ExoplanetsDialog.cpp:272]: Upgrade links to https where available.

**Comment by alex-w** [comment]
> Why not replace by QtChart immediately? This could probably re-use most of the plotting code.

I have no time for this work currently :(

**Review by gzotti — CHANGES_REQUESTED** [review]
Seriously, we should re-use the existing code and move to QChart. I think it helps during migration to set a QChartView side-by-side with the old plot and develop until both look roughly the same. The major difference is that in the old version you set X values and Y values independently, while in the QChart you use QXYSeries that would have to be created on the fly from the various already existing lists.

One point I just found: year of discovery is apparently 0 for at least one entry, making the graph almost useless. Force minimum of 1995? Or better remove entry with unknown year from graph.

**Comment by gzotti** [comment]
> > Why not replace by QtChart immediately? This could probably re-use most of the plotting code.
> 
> I have no time for this work currently :(

OK. Maybe I can have a look into this then, and may open my own replacement PR soon. However I just started the AstroCalc changes, these would be more important, and maybe you can share some insights there. 

It would be a pity to simply delete the in-fact interesting diagrams without proper replacement.

**Comment by alex-w** [comment]
OK, let's postpone this request right now for some future review or replace

**Pull request closed without merging**